.github/workflows/tests.yml

```diff
@@ -140,6 +140,19 @@ jobs:
           poetry install
           pytest -s tests
 
+  test-typegraph2:
+    needs:
+      - changes
+    if: ${{ needs.changes.outputs.typegraph == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v2
+        with:
+          shared-key: ${{ runner.os }}-rust-${{ hashFiles('**/rust-toolchain.toml') }}-${{ hashFiles('**/Cargo.lock') }}
+      - run: cargo test --package typegraph_core -- --test-threads=1
```

Comment:
multi-thread breaks?

Reply:
Yes. All the types are stored in a global store, so the indexes are different each time if other threads register some types at the same time.

Reply:
I think we could use thread_local for the global store instead of Mutex. I think it is not likely for a user to generate a typegraph from different threads. WDYT?

Reply:
Good idea!